code/modules/vehicles/mecha/equipment/weapons/weapons.dm

```diff
@@ -540,3 +540,78 @@
 	equip_cooldown = 60
 	det_time = 20
 	mech_flags = EXOSUIT_MODULE_HONK
+
+///long claw of the law
+/obj/item/mecha_parts/mecha_equipment/weapon/paddy_claw
+	name = "hydraulic claw"
+	desc = "A modified hydraulic clamp, for use exclusively with the Paddy exosuit. Non-lethally apprehends suspects."
+	icon_state = "paddy_claw"
+	equip_cooldown = 15
+	energy_drain = 10
+	tool_behaviour = TOOL_RETRACTOR
+	range = MECHA_MELEE
+	toolspeed = 0.8
+	mech_flags = EXOSUIT_MODULE_PADDY
+	///Var for the container object inside the mech
+	var/obj/item/mecha_parts/mecha_equipment/ejector/seccage/cargo_holder
+	///Audio for using the hydraulic clamp
+	var/clampsound = 'sound/mecha/hydraulic.ogg'
+	///Var for the cuff type. Basically stole how cuffing works from secbots
+	var/cuff_type = /obj/item/restraints/handcuffs/cable/zipties/used
+	///Var for autocuff, can be toggled in the mech interface.
+	var/autocuff = TRUE
+
+/obj/item/mecha_parts/mecha_equipment/weapon/paddy_claw/attach(obj/vehicle/sealed/mecha/new_mecha)
+	. = ..()
+	if(istype(chassis, /obj/vehicle/sealed/mecha/ripley/paddy))
+		cargo_holder = locate(/obj/item/mecha_parts/mecha_equipment/ejector/seccage) in chassis.equip_by_category[MECHA_UTILITY]
+
+/obj/item/mecha_parts/mecha_equipment/weapon/paddy_claw/action(mob/living/source, atom/target, list/modifiers)
+	if(!cargo_holder)
+		cargo_holder = locate(/obj/item/mecha_parts/mecha_equipment/ejector/seccage) in chassis.equip_by_category[MECHA_UTILITY]
```

Comment:
Same here, we should just eliminate this possibility in `attach` and stack trace there rather than here.

Reply:
This check also is here in case the cargo holder equipment is ever destroyed. It's not supposed to be, but y'know.